CMakeLists.txt

```diff
@@ -25,49 +51,31 @@ STRING(COMPARE EQUAL ${CMAKE_BUILD_TYPE} "DEBUG" BUILD_DEBUG)
 option(GSIBUILD "Build only libs for GSI" ON)
 
 set(definitions "UNDERSCORE" "NFILES=32" "MAXCD=250" "MAXNC=600" "MXNAF=3")
-
-if( (BUILD_RELEASE) OR (BUILD_PRODUCTION) )
-  if(IntelComp)
-    #shared with C and Fortran
-    set(shared_flags "-g" "-traceback" "-O3" "-axCORE-AVX2")
-
-    set(c_DA_allocation_def "DYNAMIC_ALLOCATION")
-    set(c_4_DA_flags)    
-    set(c_8_DA_definitions "F77_INTSIZE_8")
-    set(c_d_DA_flags)
-    
-    set(fortran_4_DA_flags)
-    set(fortran_8_DA_flags "-r8" "-i8")
-    set(fortran_d_DA_flags "-r8")
-    
-    set(c_nonDA_allocation_def "STATIC_ALLOCATION")
-    set(c_4_flags)
-    set(c_8_flags)
-    set(c_8_definitions "F77_INTSIZE_8")
-    set(c_d_flags)
-    set(c_SS_flags "-mcmodel=medium" "-shared")
-
-    set(fortran_4_flags)
-    set(fortran_8_flags "-r8" "-i8")
-    set(fortran_d_flags "-r8")
-    set(fortran_SS_flags "-mcmodel=medium" "-shared")
-
-  elseif(GNUComp)
+set(c_DA_allocation_def "DYNAMIC_ALLOCATION")
+set(c_nonDA_allocation_def "STATIC_ALLOCATION")
+
+set(c_4_DA_flags "-DUNDERSCORE")
+set(c_8_DA_definitions "F77_INTSIZE_8")
+set(c_d_DA_flags "-DUNDERSCORE")
+set(fortran_4_DA_flags)
+set(c_4_flags  "-DUNDERSCORE")
+set(c_8_flags  "-DUNDERSCORE")
+set(c_8_definitions "F77_INTSIZE_8")
+set(c_d_flags "-DUNDERSCORE")
+set(fortran_4_flags)
+if(IntelComp)
```

Comment:
Can this `if(IntelComp)` and the other compiler `comp`s general be removed and move this into the `if` at the top where you detect the compiler?

Reply:
I guess you do need to know the compiler later on.